### PR TITLE
Georgia low income credits

### LINKED ISSUE #2414: Georgia low income credit
file:///Users/eleanorzhu/Downloads/2022%20IT-511_Individual_Income_Tax_%20Booklet%20(1).pdf

### PR DESCRIPTION
**Pull request by Eleanor-Zhu**
Fixes #2414

### Summary
🧑‍👧‍👦🧓🏛️

This pull request adds new parameter files for Georgia and Oregon state income tax credits and subtractions. These parameters enable the calculation of state tax liabilities based on federal tax credits and AGI.

> _`parameters` grow_
> _adding tax credits and caps_
> _autumn of updates_

### Walkthrough
*  Add parameter files for Georgia and Oregon tax credits and subtractions ([link](https://github.com/PolicyEngine/policyengine-us/pull/2415/files?diff=unified&w=0#diff-1b82a5c84b8b6b2e4b2457da80d4497f2a88f27de0f55cb56b0b85c653b4a336R1-R43), [link](https://github.com/PolicyEngine/policyengine-us/pull/2415/files?diff=unified&w=0#diff-5ba544e740d833b7bf32d6b02f6c654ffa3348126f73fff7f300f5b65fde2fd5R1-R8), [link](https://github.com/PolicyEngine/policyengine-us/pull/2415/files?diff=unified&w=0#diff-41d72ed1bbfbbe3334a446814c9cf792924b70378da3755abf5313597b90a76fL1-R12))
  - Define Oregon federal tax liability subtraction cap for married filing separately filers in `amount.yaml` ([link](https://github.com/PolicyEngine/policyengine-us/pull/2415/files?diff=unified&w=0#diff-1b82a5c84b8b6b2e4b2457da80d4497f2a88f27de0f55cb56b0b85c653b4a336R1-R43))
  - Define age threshold for elderly or disabled credit eligibility in `supplement_age_eligibility.yaml` ([link](https://github.com/PolicyEngine/policyengine-us/pull/2415/files?diff=unified&w=0#diff-5ba544e740d833b7bf32d6b02f6c654ffa3348126f73fff7f300f5b65fde2fd5R1-R8))
  - Define Georgia tax credits for qualified child and dependent care expenses in `dependent_care_expenses.yaml` ([link](https://github.com/PolicyEngine/policyengine-us/pull/2415/files?diff=unified&w=0#diff-41d72ed1bbfbbe3334a446814c9cf792924b70378da3755abf5313597b90a76fL1-R12))

### CODE REVIEW

File: policyengine_us/parameters/gov/states/ga/tax/income/credits/low_income_credit/amount.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+description: credit that corresponds to federal adjusted gross income for georgia
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ga/tax/income/credits/low_income_credit/amount.yaml:1]:
```suggestion
description: Georgia provides the following low income tax credit amount based on their agi.
```

File: policyengine_us/parameters/gov/states/ga/tax/income/credits/low_income_credit/amount.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+metadata:
+  type: single_amount
+  threshold_unit: currency-USD
+  amount_unit: currency-USD
+  label: credit that corresponds to federal adjusted gross income for georgia
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ga/tax/income/credits/low_income_credit/amount.yaml:6]:
```suggestion
  label: Georgia low income tax credit amount
```

File: policyengine_us/parameters/gov/states/ga/tax/income/credits/low_income_credit/amount.yaml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  reference:
+    - title: georgia low income credit wooksheet(page22)
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ga/tax/income/credits/low_income_credit/amount.yaml:8]: add legal reference

File: policyengine_us/parameters/gov/states/ga/tax/income/credits/low_income_credit/amount.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+      href: https://dor.georgia.gov/document/document/2022-it-511-individual-income-tax-booklet/download
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ga/tax/income/credits/low_income_credit/amount.yaml:9]: add page

File: policyengine_us/parameters/gov/states/ga/tax/income/credits/low_income_credit/supplement_age_eligibility.yaml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+description: Age threshold to be eligible for a GA income tax aged exemption
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ga/tax/income/credits/low_income_credit/supplement_age_eligibility.yaml:1]:
```suggestion
description: Georgia allows filers to claim the aged exemption at or above this age.
```

File: policyengine_us/parameters/gov/states/ga/tax/income/credits/low_income_credit/amount.yaml [hunk 6]
```diff
@@ -0,0 +1,34 @@
+description: Georgia provides the following low income tax credit amount based on their agi.
+metadata:
+  type: single_amount
+  threshold_unit: currency-USD
+  amount_unit: currency-USD
+  label: Georgia low income tax credit amount
+  reference:
+    - title: georgia low income credit wooksheet(page22)
+      href: https://advance.lexis.com/documentpage/?pdmfid=1000516&crid=0d6fd3c6-6e65-4928-9679-0b33bed98769&config=00JAA1MDBlYzczZi1lYjFlLTQxMTgtYWE3OS02YTgyOGM2NWJlMDYKAFBvZENhdGFsb2feed0oM9qoQOMCSJFX5qkd&pddocfullpath=%2Fshared%2Fdocument%2Fstatutes-legislation%2Furn%3AcontentItem%3A6348-G0H1-DYB7-W3SK-00008-00&pdcontentcomponentid=234186&pdteaserkey=sr0&pditab=allpods&ecomp=7s65kkk&earg=sr0&prid=8be67d32-0de2-4977-a057-d9d01bc94c88
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ga/tax/income/credits/low_income_credit/amount.yaml:9]: add tax form for 2022 and 2021 tax forms - should be 3 references total (legal, 2022 form, 2021 form)

File: policyengine_us/parameters/gov/states/ga/tax/income/credits/low_income_credit/supplement_age_eligibility.yaml [hunk 7]
```diff
@@ -0,0 +1,8 @@
+description: Georgia allows filers to claim the aged exemption at or above this age.
+values:
+  1986-01-01: 65
+metadata:
+  unit: year
+  reference:
+    - title: low income credit wooksheet(page22)
+      href: https://advance.lexis.com/documentpage/?pdmfid=1000516&crid=0d6fd3c6-6e65-4928-9679-0b33bed98769&config=00JAA1MDBlYzczZi1lYjFlLTQxMTgtYWE3OS02YTgyOGM2NWJlMDYKAFBvZENhdGFsb2feed0oM9qoQOMCSJFX5qkd&pddocfullpath=%2Fshared%2Fdocument%2Fstatutes-legislation%2Furn%3AcontentItem%3A6348-G0H1-DYB7-W3SK-00008-00&pdcontentcomponentid=234186&pdteaserkey=sr0&pditab=allpods&ecomp=7s65kkk&earg=sr0&prid=8be67d32-0de2-4977-a057-d9d01bc94c88
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ga/tax/income/credits/low_income_credit/supplement_age_eligibility.yaml:8]: add tax form for 2022 and 2021 tax forms - should be 3 references total (legal, 2022 form, 2021 form)

File: policyengine_us/parameters/gov/states/ga/tax/income/credits/low_income_credit/amount.yaml [hunk 8]
```diff
@@ -0,0 +1,38 @@
+description: Georgia provides the following low income tax credit amount based on their agi.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ga/tax/income/credits/low_income_credit/amount.yaml:1]:
```suggestion
description: Georgia provides the following low income tax credit amount based on agi.
```

File: policyengine_us/parameters/gov/states/ga/tax/income/credits/low_income_credit/supplement_age_eligibility.yaml [hunk 9]
```diff
@@ -0,0 +1,12 @@
+description: Georgia allows filers to claim the aged exemption at or above this age.
+values:
+  1986-01-01: 65
+metadata:
+  unit: year
+  reference:
+    - title: ga allows filers to claim the aged exemption at or above this age | O.C.G.A. § 48-7A-3
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ga/tax/income/credits/low_income_credit/supplement_age_eligibility.yaml:7]:
```suggestion
    - title: O.C.G.A. § 48-7A-3
```

File: policyengine_us/parameters/gov/states/ga/tax/income/credits/low_income_credit/amount.yaml [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+metadata:
+  type: single_amount
+  threshold_unit: currency-USD
+  amount_unit: currency-USD
+  label: Georgia low income tax credit amount
+  reference:
+    - title: ga allows filers to claim the aged exemption at or above this age | O.C.G.A. § 48-7A-3
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ga/tax/income/credits/low_income_credit/amount.yaml:8]:
```suggestion
    - title:  O.C.G.A. § 48-7A-3
```

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_us/parameters/gov/states/ga/tax/income/credits/low_income/amount.yaml [hunk 11]
```diff
@@ -0,0 +1,38 @@
+description: Georgia provides the following low income tax credit amount based on agi.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/ga/tax/income/credits/low_income/amount.yaml:1]:
```suggestion
description: Georgia provides the following low income tax credit amount based on adjusted gross income.
```

File: policyengine_us/parameters/gov/states/ga/tax/income/credits/low_income/supplement_age_eligibility.yaml [hunk 12]
```diff
@@ -0,0 +1,13 @@
+description: Georgia provides an additional exemption under the low income credit for head and spouse at or above this age threshold.
+values:
+  2021-01-01: 65
+metadata:
+  unit: year
+  label: Georgia low income credit supplement age eligibility
+  reference:
+    - title: Official Code of Georgia Annotated | CH17 Income Taxes | O.C.G.A. § § 48-7A-3 (b)
+      href: https://advance.lexis.com/documentpage/?pdmfid=1000516&crid=0d6fd3c6-6e65-4928-9679-0b33bed98769&config=00JAA1MDBlYzczZi1lYjFlLTQxMTgtYWE3OS02YTgyOGM2NWJlMDYKAFBvZENhdGFsb2feed0oM9qoQOMCSJFX5qkd&pddocfullpath=%2Fshared%2Fdocument%2Fstatutes-legislation%2Furn%3AcontentItem%3A6348-G0H1-DYB7-W3SK-00008-00&pdcontentcomponentid=234186&pdteaserkey=sr0&pditab=allpods&ecomp=7s65kkk&earg=sr0&prid=8be67d32-0de2-4977-a057-d9d01bc94c88
+    - title: 2022 georgia low income credit worksheet, page22
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/ga/tax/income/credits/low_income/supplement_age_eligibility.yaml:10]: And elsewhere
```suggestion
    - title: 2022 Georgia Low Income Credit Worksheet, page 22
```

File: policyengine_us/tests/policy/baseline/gov/states/ga/tax/income/credits/ga_low_income_credit.yaml [hunk 13]
```diff
@@ -0,0 +1,65 @@
+- name: Georgia low income credits higher income bracket
+  period: 2022
+  input:
+    state_code: GA
+    adjusted_gross_income: 14_000
+    age_head: 67
```
Comment (MaxGhenis) [inline policyengine_us/tests/policy/baseline/gov/states/ga/tax/income/credits/ga_low_income_credit.yaml:6]: Please test the edge cases, 65 and 64

File: policyengine_us/variables/gov/states/ga/tax/income/credits/ga_low_income_credit.py [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+    value_type = float
+    entity = TaxUnit
+    label = "Georgia low income credit"
+    unit = USD
+    definition_period = YEAR
+    reference = "https://dor.georgia.gov/document/document/2022-it-511-individual-income-tax-booklet/download"
+    defined_for = StateCode.GA
+
+    def formula(tax_unit, period, parameters):
+        # federal ajusted gross income
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/ga/tax/income/credits/ga_low_income_credit.py:14]: Reorganize the code to follow the steps from the form and legal code:
1. Count exemptions
2. Add aged exemptions
3. Get amount per exemption
4. Return the product

Only get the parameters and AGI when you need them.

File: policyengine_us/variables/gov/states/ga/tax/income/credits/ga_low_income_credit.py [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+            int
+        )  # if so, return 1, otherwise return 0
+        aged_spouse = (tax_unit("age_spouse", period) >= age_threshold).astype(
+            int
+        )  # if so, return 1, otherwise return 0
+        aged_count = aged_head + aged_spouse
+        exemptions = tax_unit("exemptions", period)
+        total_exemptions = aged_count + exemptions
+        # amount threshold
+        amount_per_aged = p.amount.calc(federal_agi)
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/ga/tax/income/credits/ga_low_income_credit.py:29]:
```suggestion
        amount_per_exemption = p.amount.calc(federal_agi)
```

File: policyengine_us/variables/gov/states/ga/tax/income/credits/ga_low_income_credit.py [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+        # age threshold
+        age_threshold = p.supplement_age_eligibility
+        aged_head = (tax_unit("age_head", period) >= age_threshold).astype(
+            int
+        )  # if so, return 1, otherwise return 0
+        aged_spouse = (tax_unit("age_spouse", period) >= age_threshold).astype(
+            int
+        )  # if so, return 1, otherwise return 0
+        aged_count = aged_head + aged_spouse
+        exemptions = tax_unit("exemptions", period)
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/ga/tax/income/credits/ga_low_income_credit.py:26]:
```suggestion
        # We follow the legal code, which says (in addition to head and spouse):
        # "multiplied by the number of dependents which the taxpayer is entitled to claim."
        # The tax form excludes adult dependents:
        # "Exemptions are self, spouse and natural or legally adopted children"
        exemptions = tax_unit("exemptions", period)
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

**Review by MaxGhenis — APPROVED** [review]
